### ISSUE TITLE
Wrong type in deployment properties

### ISSUE THREAD

**Issue by pierskarsenbarg** [pulumi/pulumi-hugo#392]
File: [themes/default/content/docs/reference/pkg/azure-native/resources/deployment.md](https://www.pulumi.com/docs/reference/pkg/azure-native/resources/deployment/)

In the example at the top of the page, it shows the `mode` input as a string, but it should be an enum.

Current:

```csharp
using Pulumi;
using AzureNative = Pulumi.AzureNative;

class MyStack : Stack
{
    public MyStack()
    {
        var deployment = new AzureNative.Resources.Deployment("deployment", new AzureNative.Resources.DeploymentArgs
        {
            DeploymentName = "my-deployment",
            Properties = new AzureNative.Resources.Inputs.DeploymentPropertiesArgs
            {
                Mode = "Incremental",
                Parameters = ,
                TemplateLink = new AzureNative.Resources.Inputs.TemplateLinkArgs
                {
                    QueryString = "sv=2019-02-02&st=2019-04-29T22%3A18%3A26Z&se=2019-04-30T02%3A23%3A26Z&sr=b&sp=rw&sip=168.1.5.60-168.1.5.70&spr=https&sig=xxxxxxxx0xxxxxxxxxxxxx%2bxxxxxxxxxxxxxxxxxxxx%3d",
                    Uri = "https://example.com/exampleTemplate.json",
                },
            },
            ResourceGroupName = "my-resource-group",
        });
    }

}
```

Should be:  

```csharp
using Pulumi;
using AzureNative = Pulumi.AzureNative;

class MyStack : Stack
{
    public MyStack()
    {
        var deployment = new AzureNative.Resources.Deployment("deployment", new AzureNative.Resources.DeploymentArgs
        {
            DeploymentName = "my-deployment",
            Properties = new AzureNative.Resources.Inputs.DeploymentPropertiesArgs
            {
                Mode = AzureNative.Resources.DeploymentMode.Incremental,
                Parameters = ,
                TemplateLink = new AzureNative.Resources.Inputs.TemplateLinkArgs
                {
                    QueryString = "sv=2019-02-02&st=2019-04-29T22%3A18%3A26Z&se=2019-04-30T02%3A23%3A26Z&sr=b&sp=rw&sip=168.1.5.60-168.1.5.70&spr=https&sig=xxxxxxxx0xxxxxxxxxxxxx%2bxxxxxxxxxxxxxxxxxxxx%3d",
                    Uri = "https://example.com/exampleTemplate.json",
                },
            },
            ResourceGroupName = "my-resource-group",
        });
    }

}
```

**Comment by joeduffy** [comment]
@viveklak Is this a bug in arm2pulumi, in that it doesn't know about enums?

Or perhaps in the central code generators?

Do we have similar issues in code generated by tools like tf2pulumi, the import command, etc?

**Comment by viveklak** [comment]
> @viveklak Is this a bug in arm2pulumi, in that it doesn't know about enums?
> 
> Or perhaps in the central code generators?
> 
> Do we have similar issues in code generated by tools like tf2pulumi, the import command, etc?

Program gen currently doesn't honor strict enums, see https://github.com/pulumi/pulumi/issues/6263. It is limited to azure-native and kubernetes since we actually model strict enums in these schemas while the bridged providers only support relaxed enums (so arm2pulumi is uniquely affected by this).

**Comment by joeduffy** [comment]
Gotcha, should this be closed as a dupe of https://github.com/pulumi/pulumi/issues/6263?

**Comment by viveklak** [comment]
> Gotcha, should this be closed as a dupe of [pulumi/pulumi#6263](https://github.com/pulumi/pulumi/issues/6263)?

Yup. Closing in favor of https://github.com/pulumi/pulumi/issues/6263
